pkg/auth/defaultmanager.go

```diff
@@ -0,0 +1,30 @@
+package auth
+
+import (
+	"errors"
+	"log"
+
+	"github.com/codeready-toolchain/registration-service/pkg/configuration"
+)
+
+var defaultKeyManager *KeyManager
+
+// DefaultKeyManagerWithConfig creates the default manager if it has not created yet.
```

Comment:
```suggestion
// DefaultKeyManagerWithConfig creates the default manager if it was not created yet.
```

Reply:
This is all a bit bad. I removed the `sync.Do`, because the impl from the legacy fabric8 code could silently fail there when called more than once. Found a better solution. Fixed 8954a70.

Reply:
could you elaborate on the failure you mentioned, please? AFAIK, `sync.Once` is what we need here

Reply:
`sync.Once` is at least not enough as a repeated call to `DefaultKeyManagerWithConfig()` with an *updated* config (or logger) would just silently ignore the new config and return the existing instance. I'd say that is bad design as the caller don't get any indication that his call (with a different config) is returning an instance with the "old" config. If we don't want to have this called more than once, then let's drop an error when it is called more than once (or make it recognize updated arguments, which is not an option here).

Just returning something that does ignore the arguments will probably lead to hard to track down issues where a caller expects the config to be recognized if the call does return without an error.

And if we just check the instance existence, we don't need the `sync.Once` here.

Reply:
ok, I see what you mean with the config update use-case, now. But should we use a lock to avoid concurrent init, at least?

Reply:
Ok, Tina and I found a better alternative that has the issue solved while being thread safe. See 59c23be.